### PR TITLE
Implement ExceptionSpec

### PR DESCRIPTION
**Pull request by kriskowal**
r @jcorbin @malandrew @leizha

### CODE REVIEW

File: exception.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    assert(typeof def.annotations.type === 'string',
+        'type annotation must be a string: ' + self.name);
+    assert(typeof def.annotations.message === 'string',
+        'message annotation must be a string: ' + self.name);
+    self.type = def.annotations.type;
+    self.message = def.annotations.message;
+};
+
+ExceptionSpec.prototype.createConstructor =
+function createConstructor(name, fieldNames) {
```
Comment (jcorbin) [inline exception.js]: smell, this is duplicated with structspec right? not the worst sin, but would be nice to undupe it some day

File: test/exception.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+var path = require('path');
+var Spec = require('../spec');
+
+var source = fs.readFileSync(path.join(__dirname, 'exception.thrift'), 'ascii');
+var spec = new Spec({source: source});
+
+test('Exception RW', testRW.cases(spec.BogusNameError.rw, [
+
+    [spec.BogusNameError({bogusName: 'Voldemort'}), [
+        0x0b, // typeid:1 = 11, STRING
```
Comment (jcorbin) [inline test/exception.js]: please wrap and align these like the other tests

**Comment by jcorbin** [comment]
needs a rebase, and please keep the fixtures consistent, it can be easy:
- install https://github.com/godlygeek/tabular plugin
- with cursor on any one of the fixtures lines, `:Tab/\/\/`
- add normal mappings to taste for #productivity

**Comment by jcorbin** [comment]
oh but: lgtm otherwise

File: exception.js (hunk 1 above)
Comment (kriskowal) [inline exception.js]: This is an override, not a dup. Struct createConstructor builds a function through eval. This creates a TypedError instead.

**Pull request merged**